db/seeds.dev.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+FactoryBot.create_list(:account, 3)
+
+Account.find_each do |account|
+  host_ids = SupportedSsg.all.map { |s| [s.os_major_version, s.os_minor_version] }.uniq.flat_map do |version|
+    FactoryBot.create_list(:host, rand(2..3), account: account.account_number, os_version_arr: version).map(&:id)
+  end
+
+  Profile.canonical.includes(:benchmark).group_by(&:ref_id).each do |_, profiles|
+    canonical_profile = profiles.sample
+    hosts = Host.where(id: host_ids).where("#{Host::OS_MAJOR_VERSION} = ?", canonical_profile.os_major_version)
+
+    next if hosts.empty? || [true, true, false].sample
+
+    profile = Profile.new(parent_profile_id: canonical_profile.id, account: account).fill_from_parent
+    policy = Policy.new(profile.slice(:name, :description, :compliance_threshold, :account))
+    policy.save!
+    profile.update!(policy: policy)
+
+    policy.update_hosts(hosts.pluck(:id))
+
+    passing_hosts = policy.hosts.sample(rand(1..policy.hosts.count / 2))
+
+    hosts.find_each do |host|
+      next if [true, true, false].sample
+
+      profile = policy.reload.profiles.find_by(os_minor_version: host.os_minor_version)
+      profile ||= canonical_profile.clone_to(
+        policy: policy,
+        account: account,
+        os_minor_version: host.os_minor_version.to_s
+      )
+
+      test_results = FactoryBot.create_list(
+        :test_result,
+        rand(2..3),
+        profile: profile,
+        host: host,
+        supported: SupportedSsg.supported?(
+          ssg_version: profile.ssg_version,
+          os_major_version: host.os_major_version,
+          os_minor_version: host.os_minor_version
+        )
+      )
+
+      test_results.each do |tr|
+        rule_result_columns = profile.rules.pluck(:id).map do |rule_id|
+          [rule_id, tr.id, host.id, passing_hosts.include?(host) ? 'pass' : %w[pass pass fail].sample]
+        end
+
+        RuleResult.import(%i[rule_id test_result_id host_id result], rule_result_columns)
+
+        tr.update!(score: tr.rule_results.passed.count / tr.rule_results.count.to_f * 100)
```

Comment:
Isn't there a method to calculate this?

Reply:
Nope, this is coming directly from the XML file